.github/workflows/release-lib.yaml

```diff
@@ -0,0 +1,124 @@
+        uses: actions/checkout@v2
+        with:
+          ref: refs/heads/${{github.event.workflow_run.head_branch}}
+      - name: Get tag
+        run: |
+          branch=${{github.event.workflow_run.head_branch}}
+          echo '::set-output name=tag::'${branch#release/}
+        id: get-tag-step
+      - name: Check non-release dependencies
+        run: |
+          nonrelease=$(grep < go.mod -P \
+            '^\tgithub.com/networkservicemesh/(?!gotestmd )[a-zA-Z0-9_-]+ v[^ ]+[.-][0-9]{14}-[0-9a-f]{12}' || true)
+          if [ "$nonrelease" ]; then
+            echo "Found non-release dependencies:"
+            echo "$nonrelease"
+            ## exit 1
+          fi
+      - name: Push tag ${{ steps.get-tag-step.outputs.tag }}
+        run: |
+          git status
+          git tag ${{ steps.get-tag-step.outputs.tag }}
+          git push origin ${{ steps.get-tag-step.outputs.tag }} -f
+      - name: Create release ${{ steps.get-tag-step.outputs.tag }}
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.NSM_BOT_GITHUB_TOKEN }}
+        with:
+          tag_name: refs/tags/${{ steps.get-tag-step.outputs.tag }}
+          release_name: ${{ steps.get-tag-step.outputs.tag }}
+          draft: false
+          prerelease: false
+
+  update-dependent-repositories:
+    strategy:
+      matrix:
+        repository: ${{fromJson(inputs.dependent_repositories)}}
+    name: Update ${{ matrix.repository }}
+    needs: create-release
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success' && contains(github.event.workflow_run.head_branch, 'release/') && inputs.dependent_repositories }}
+    steps:
+      - name: Get tag
+        run: |
+          branch=${{github.event.workflow_run.head_branch}}
+          echo '::set-output name=tag::'${branch#release/}
+        id: get-tag-step
+      - name: Create commit
+        run: |
+          echo "Update go.mod and go.sum to ${{ github.repository }}@${{ steps.get-tag-step.outputs.tag }}" >> /tmp/commit-message
+      - name: Checkout networkservicemesh/${{ matrix.repository }}
+        uses: actions/checkout@v2
+        with:
+          path: networkservicemesh/${{ matrix.repository }}
+          repository: networkservicemesh/${{ matrix.repository }}
+          token: ${{ secrets.NSM_BOT_GITHUB_TOKEN }}
+      - uses: actions/setup-go@v1
+        with:
+          go-version: 1.16
+      - name: Update ${{ github.repository }}
+        working-directory: networkservicemesh/${{ matrix.repository }}
+        run: |
+          git config --global user.email "nsmbot@networkservicmesh.io"
+          git config --global user.name "NSMBot"
+
+          release=${{ github.event.workflow_run.head_branch }}
+
+          for attempt in {1..5}; do
+            echo "attempt $attempt"
+            git fetch -p origin
+            base=$release
+            git show-ref --quiet refs/remotes/origin/$base || base=main
```

Comment:
note: base=main is not expected for release.

Reply:
If the `$release` branch doesn't exist, we start it from the `main` branch. Or should we use a specific starting commit?

Reply:
I think we should stop release if we don't have release branch

Reply:
It's a branch in the dependent repository (not in the current one, which is running on the CI). Currently, branches in dependent repositories are created automatically from the `main` branch.

Details: our current `release.yaml` [uses](https://github.com/networkservicemesh/sdk-sriov/blob/9efc2a84b057be9ddfef30982ada07ee0e79e343/.github/workflows/release.yaml#L67-L72) the `actions/checkout` action to fetch the dependent repository. According to [`actions/checkout` doc](https://github.com/actions/checkout/blob/v2/README.md?plain=1#L45), the default (i.e. `main`) branch is used. Then our `release.yaml` [creates](https://github.com/networkservicemesh/sdk-sriov/blob/9efc2a84b057be9ddfef30982ada07ee0e79e343/.github/workflows/release.yaml#L90-L91) a release branch, dropping an already existing branch (since the force push `-f` flag being used).

---

In short, the current logic of `update-dependent-repositories` job:
Create a release branch starting from the `main` branch. Override it if it already exists (hence the conflict described in the issue).

Logic In this PR:
Create a release branch if it doesn't exist. Otherwise, append a commit to an existing release branch.